src/packet_router/mod.rs

```diff
@@ -172,85 +163,27 @@ impl Router {
     }
 
     async fn send_waiting_packets(&mut self, logger: &Logger) {
-        while let Some(packet) = self.store.pop_waiting_packet() {
+        while let Some(packet) = self.store.pop_waiting() {
             match self.send_packet(logger, packet).await {
                 Ok(()) => (),
                 Err(err) => warn!(logger, "failed to send uplink {err:?}"),
             }
         }
     }
 
-    pub async fn mk_uplink(&self, packet: QuePacket) -> Result<PacketRouterPacketUpV1> {
-        let region = self.region.ok_or_else(RegionError::no_region_params)?;
-
-        let mut uplink = PacketRouterPacketUpV1::try_from(packet)?;
-        uplink.region = region.into();
+    pub async fn mk_uplink(&self, packet: CacheMessage<Packet>) -> Result<PacketRouterPacketUpV1> {
+        let mut uplink: PacketRouterPacketUpV1 = packet.into_inner().try_into()?;
+        uplink.region = self.region_params.region.into();
         uplink.gateway = self.keypair.public_key().into();
         uplink.signature = uplink.sign(self.keypair.clone()).await?;
         Ok(uplink)
     }
 
-    async fn send_packet(&mut self, logger: &Logger, packet: QuePacket) -> Result {
+    async fn send_packet(&mut self, logger: &Logger, packet: CacheMessage<Packet>) -> Result {
```

Comment:
Why take the wrapped packet? I feel like `CacheMessage` should be an internal detail to the cache.

Reply:
Because `hold_time` comes out of the CacheMessage